src/expr/syntax_tree.rs

```diff
@@ -85,15 +85,35 @@ impl ASTNode {
             ASTNode::Node { ref op_type, .. } => match self.operand_values() {
                 Err(reason) => Err(reason),
                 Ok(operand_values) => match op_type.as_ref() {
-                    "+" => infix_operator_two_ints(|a: i64, b: i64| Ok(a + b), &operand_values),
-                    "-" => infix_operator_two_ints(|a: i64, b: i64| Ok(a - b), &operand_values),
-                    "*" => infix_operator_two_ints(|a: i64, b: i64| Ok(a * b), &operand_values),
-                    "/" => infix_operator_two_ints(
-                        |a: i64, b: i64| {
+                    "+" => infix_operator_two_ints(|a: i64, b: i64| {
+                            match a.checked_add(b) {
+                                Some(v) => Ok(v),
+                                None => Err("+: Numerical result out of range".to_owned()),
+                            }
+                        }, &operand_values
+                    ),
+                    "-" => infix_operator_two_ints(|a: i64, b: i64| {
+                            match a.checked_sub(b) {
```

Comment:
I'd prefer that these use a function or something to reduce the boilerplate:
```rust
fn checked_binop<F: Fn() -> Option<T>, T>(cb: F, op: &str) -> Result<T, String> {
    match cb() {
        Some(v) => Ok(v),
        None => Err(format!("{}: Numerical result out of range", op)),
    }
}
```

Then, for example, "+" would be:
```rust
"+" => infix_operator_two_ints(|a: i64, b: i64| {
        checked_binop(|| a.checked_add(b))
    }, &operand_values)
```